### PR TITLE
fix(translations): add XmTranslationModule

### PR DESCRIPTION
**Pull request by Kaskyi**

### CODE REVIEW

**Comment by Kaskyi** [comment]
- Add JHipster directive and pipes into xm-webapp as Jhi module;
  Because NgJhipsterModule implements ngx-translate in a specific way, which effects on the whole system, Ive removed Module and added components directly. Alternative solution could be  `i18nEnabled: false;` but it does not work well.
- Remove NgJhipsterModule;
- Add LanguageService;
- Update tslint config;
- Add Title service for the document.title;
- languageHelper and modulesLangHelper refactoring;

File: src/app/application/entity-detail.component.ts [hunk 1]
```diff
@@ -14,17 +14,14 @@ declare var $: any;
 })
 export class EntityDetailComponent implements OnInit, OnDestroy {
 
+    xmEntity: XmEntity;
```
Comment (ishkurko) [inline src/app/application/entity-detail.component.ts]: Plz, add visibility.

File: src/app/modules/jhi/capitalize.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'capitalize' })
```
Comment (ishkurko) [inline src/app/modules/jhi/capitalize.pipe.ts]: Did you just copy this pipe or copied with modification?

File: src/app/modules/jhi/filter.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'filter', pure: false })
```
Comment (ishkurko) [inline src/app/modules/jhi/filter.pipe.ts]: Did you just copy this pipe or copied with modification?

File: src/app/modules/jhi/capitalize.pipe.ts (hunk 2 above)
Comment (Kaskyi) [reply src/app/modules/jhi/capitalize.pipe.ts]: just copy

File: src/app/modules/xm-translation/title.service.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { Title } from '@angular/platform-browser';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { TranslateService as NgxTranslateService } from '@ngx-translate/core';
+import { Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { LanguageService, Translate } from './language.service';
```
Comment (ishkurko) [inline src/app/modules/xm-translation/title.service.ts]: There is a good practice to add empty line between internal and external imports.
Look at https://angular.io/guide/styleguide

File: src/app/modules/jhi/filter.pipe.ts (hunk 3 above)
Comment (Kaskyi) [reply src/app/modules/jhi/filter.pipe.ts]: just copy

File: src/app/modules/xm-translation/title.service.ts (hunk 4 above)
Comment (Kaskyi) [reply src/app/modules/xm-translation/title.service.ts]: fixed

File: src/app/application/entity-detail.component.ts (hunk 1 above)
Comment (Kaskyi) [reply src/app/application/entity-detail.component.ts]: fixed

**Comment by ishkurko** [comment]
A lot of moved classes from JHipsters smells bad. What is your vision about the future improvement of this code?

**Comment by Kaskyi** [comment]
``"ng-jhipster": "0.11.1"`` this update has fixed the problem. I removed dublications and updated the request.

**Pull request merged**